overlord/configstate/configcore/timezone.go

```diff
@@ -0,0 +1,86 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2020 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package configcore
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"regexp"
+
+	"github.com/snapcore/snapd/osutil"
+	"github.com/snapcore/snapd/overlord/configstate/config"
+)
+
+func init() {
+	// add supported configuration of this module
+	supportedConfigurations["core.system.timezone"] = true
+}
+
+var validTimezone = regexp.MustCompile(`^[a-zA-Z0-9+_-]+(/[a-zA-Z0-9+_-]+)?(/[a-zA-Z0-9+_-]+)?$`).MatchString
+
+func validateTimezoneSettings(tr config.ConfGetter) error {
+	timezone, err := coreCfg(tr, "system.timezone")
+	if err != nil {
+		return err
+	}
+	if timezone == "" {
+		return nil
+	}
+	if !validTimezone(timezone) {
+		return fmt.Errorf("cannot set timezone %q: name not valid", timezone)
```

Comment:
Go's time module has time.LoadLocation function, not sure if it's worth it but could be used as well.
```
location, err := time.LoadLocation("America/Los_Angeles")
if err != nil {
   ...
```

Reply:
Hmm, maybe not a good idea, it probably depends on runtime stuff that we might want to avoid e.g. with early config

Reply:
Yeah, not sure about this for exactly the reason you outlined - build-host != runtime system and all that.